src/Native/Unix/System.Native/pal_io.cpp

```diff
@@ -360,6 +363,28 @@ extern "C" int32_t SystemNative_ReadDirR(DIR* dir, void* buffer, int32_t bufferS
 
     // 0 returned with non-null result (guaranteed to be set to entry arg) -> success
     assert(result == entry);
+#else
+    errno = 0;
+    result = readdir(dir);
+
+    //  kernel set errno -> failure
```

Comment:
> Do you think I should make this edit to the other #if path as well as the one I added?

No, the other path is different: readdir_r returns the error value that's being checked again.